From 5f9bf7073d784f42c59e8fda03da70e873dd94b3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Fri, 18 Oct 2024 17:00:35 +0200
Subject: [PATCH] Skip testcase which is incompatible which libxml >= 2.13.0

There is an upstream patch for this but it requires a behavioral
change, hence we don't really want to backport it for now.
---
 test/xml/test_cdata.rb | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/test/xml/test_cdata.rb b/test/xml/test_cdata.rb
index a50bce17..f1cb8ac3 100644
--- a/test/xml/test_cdata.rb
+++ b/test/xml/test_cdata.rb
@@ -29,6 +29,11 @@ describe Nokogiri::XML::CDATA do
     end
 
     it "has nil content when passed nil" do
+      # This test won't pass with libxml-2.13.0 without a behavioral change.
+      #
+      # See https://github.com/sparklemotion/nokogiri/commit/a53df25ff6d7d1d84aa3acd73618243cf260e41e
+      skip if Nokogiri.uses_libxml?(">= 2.13.0")
+
       node = Nokogiri::XML::CDATA.new(Nokogiri::XML::Document.new, nil)
 
       assert_instance_of(Nokogiri::XML::CDATA, node)
